### PR TITLE
fix: block when stream insert

### PR DESCRIPTION
**Pull request by Fengys123**

### CODE REVIEW

File: src/stream_insert.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        mut client: GreptimeDatabaseClient<Channel>,
+        dbname: String,
+        auth_header: Option<AuthHeader>,
+    ) -> StreamInsertor {
+        let (send, recv) = tokio::sync::mpsc::channel(1024);
+
+        let join: JoinHandle<std::result::Result<Response<GreptimeResponse>, Status>> =
+            tokio::spawn(async move {
+                let recv_stream = ReceiverStream::new(recv);
+                client.handle_requests(recv_stream).await
```
Comment (fengjiachun) [inline src/stream_insert.rs:65]: map to `Result<GreotimeResponse>` ?
```
client.handle_requests(receiver).await.map(|r| r.into_inner())
```

File: src/stream_insert.rs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        let response = response
+            .into_inner()
+            .response
+            .context(IllegalDatabaseResponseSnafu {
+                err_msg: "GreptimeResponse is empty",
+            })?;
+
+        let greptime_response::Response::AffectedRows(AffectedRows { value }) = response;
+
+        Ok(value)
```
Comment (fengjiachun) [inline src/stream_insert.rs:102]: I tend to directly return `greptime_response::Response`

Comment (Fengys123) [reply src/stream_insert.rs:102]: I see that Result\<u32\> is returned [here](https://github.com/GreptimeTeam/greptimedb-client-rust/blob/master/src/database.rs#L72), shouldn’t it be consistent with this?

Comment (fengjiachun) [reply src/stream_insert.rs:102]: Let's keep the same style, using `Result<u32>`.

File: src/stream_insert.rs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+///
+/// ```ignore
+/// let grpc_client = Client::with_urls(vec!["127.0.0.1:4002"]);
+/// let client = Database::new_with_dbname("db_name", grpc_client);
+/// let stream_insertor = client.streaming_insertor().unwrap();
+/// ```
+///
+/// If you want to see a concrete usage example, please see
+/// [stream_insertor.rs](https://github.com/GreptimeTeam/greptimedb-client-rust/tree/master/examples/stream_ingest.rs).
+pub struct StreamInsertor {
```
Comment (sunng87) [inline src/stream_insert.rs:44]: Inserter

**Review by sunng87 — APPROVED** [review]

**Pull request merged**